x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/data/ElementType.java

```diff
@@ -29,6 +29,11 @@ public enum ElementType {
      */
     DOC(DocBlock::newBlockBuilder),
 
+    /**
+     * Composite blocks which contain array of sub-blocks.
+     */
+    COMPOSITE((blockFactory, estimatedSize) -> { throw new UnsupportedOperationException("can't build composite blocks"); }),
```

Comment:
:+1:

I could see us wanting some kind of composite element type one day, but we can cross that bridge when we actually need it.